src/Avalonia.Controls/TabControl.cs

```diff
@@ -162,8 +164,8 @@ protected internal override void PrepareContainerForItemOverride(Control element
 
             if (index == SelectedIndex && element is ContentControl container)
             {
-                SelectedContentTemplate = container.ContentTemplate;
-                SelectedContent = container.Content;
+                SetCurrentValue(SelectedContentTemplateProperty, container.ContentTemplate);
+                SetCurrentValue(SelectedContentProperty, container.Content);
```

Comment:
Not sure about these ones either. If `SelectedContentX` is a readonly property then does it need to be styled?

Reply:
Reverted for now